src/Symfony/Bridge/Monolog/Processor/DebugProcessor.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Processor;
+
+use Monolog\Logger;
+use Symfony\Component\HttpKernel\Log\DebugLoggerInterface;
+
+class DebugProcessor implements DebugLoggerInterface
+{
+    private $records = array();
+    private $errorCount = 0;
+
+    public function __invoke(array $record)
+    {
+        $this->records[] = array(
+            'timestamp' => $record['datetime']->getTimestamp(),
+            'message' => $record['message'],
+            'priority' => $record['level'],
+            'priorityName' => $record['level_name'],
+            'context' => $record['context'],
+            'channel' => isset($record['channel']) ? $record['channel'] : '',
+        );
+        switch ($record['level']) {
```

Comment:
OK then ... but according to 3v4l.org, the `if` code generates 7 opcodes and the `switch` code generates 11 opcodes (although it's true that not all opcodes are equal!)